### ISSUE TITLE
Show progress during unfairness mitigation

### ISSUE THREAD

**Issue by romanlutz** [fairlearn/fairlearn#510]
#### Is your feature request related to a problem? Please describe.
When running `GridSearch` or `ExponentiatedGradient` it may take a while to run all the iterations. It would be nice to have an idea about the progress. 

#### Describe the solution you'd like
For `GridSearch` this could be something simple like `completed 7/100 iterations`, or perhaps something more sophisticated like `completed 7/100 iterations, projected to finish in 4:32`. For `ExponentiatedGradient` we can't know the number of iterations ahead of time, but we can tell the user how close we are to the stopping criterion as well as the current iteration number (in case we hit the maximum number of iterations).
We should perhaps also think about solutions other packages have for this.

**Comment by riedgar-ms** [comment]
As noted on the original email thread, setting the logger to 'debug' level will at least get regular output from the algorithms, so users know that _something_ is happening.

**Comment by romanlutz** [comment]
Yeah, but that doesn't tell you much. You can sort of infer that each group of log messages is an iteration, but it's impossible to keep count...

**Comment by hildeweerts** [comment]
I haven't looked into fairlearn's GridSearch implementation so I don't know whether this is relevant, but I believe that in scikit-learn's GridSearchCV progress is printed through the joblib.Parallel dependency.

**Comment by romanlutz** [comment]
@hildeweerts it’s definitely relevant. @adrinjalali suggested using GridSearchCV for this a while back, so if we make such a change we could get that perhaps for free.

**Comment by riedgar-ms** [comment]
joblib.Parallel looks like the sort of thing we should be using to speed up GridSearch training. Would be a bit of an overkill just for adding a print statement for every iteration, though.

**Comment by arjsingh** [comment]
How about using something like tqdm? 
https://tqdm.github.io/

**Comment by riedgar-ms** [comment]
That looks good as an option - please fork and PR :-)

**Comment by romanlutz** [comment]
The consensus in #517 was not to use `tqdm` to avoid introducing a new dependency. Instead, `logging` is preferable as @adrinjalali explained in #517 . If anyone wants to start with this task please comment below!

**Comment by StefanieSenger** [comment]
It seems to me that fairlearn can use scikit-learn's callback infrastructure once it's merged?

The open PR is here: scikit-learn/scikit-learn#28760

**Comment by TamaraAtanasoska** [comment]
@StefanieSenger I am not experienced with the callbacks, would we be able to have a progress bar for every estimator as long as it inherits from `CallbackSupportMixin`?

**Comment by adrinjalali** [comment]
The details are still in the works, but all estimators will be able to provide a way for the user to get progress bars. The work has been a bit stalled for the last couple of months, but it's in our plans to move it forward.
